batch/batch/front_end/front_end.py

```diff
@@ -549,15 +561,41 @@
             assert len(rows) == 0
             raise web.HTTPForbidden(reason=f'unknown billing project {billing_project}')
 
+        maybe_batch = await tx.execute_and_fetchone(
+            '''
+SELECT * FROM batches
+WHERE token = %s AND user = %s;
+''',
+            (token, user))
+
+        if maybe_batch is not None:
```

Comment:
My feeling is this is more trouble than it's worth.  If you retry a create with the same token but different specs, I'm 100% happy to give you the initial job.

Reply:
removed